src/build_packages.sh

```diff
@@ -2,27 +2,65 @@
 set -e
 export XROAD=$(cd "$(dirname "$0")"; pwd)
 
-if [[ ! " $* " =~ " --packageonly " ]]; then
-  ./compile_code.sh "$@"
-fi
+HAS_DOCKER=""
+
+errorExit() {
+    echo "*** $*" 1>&2
+    exit 1
+}
+
+usage () {
+    echo "Usage: $0 [option for $0...] [other options]"
+    echo "Options for $0:"
+    echo " -p, --package-only    Skip compilation, just build packages'"
+    echo " -d, --docker-compile  Compile in docker container instead of native gradle build"
+    echo " -h, --help            This help text."
+    echo "The option for $0, if present, must come fist, before other options."
+    echo "Other options are passed on to compile_code.sh"
+    test -z "$1" || exit "$1"
+}
+
+buildInDocker() {
+    test -n "$HAS_DOCKER" || errorExit "Error, docker is not installed/running."
+    echo "Building in docker..."
+    docker build -q -t xroad-build --build-arg uid=$(id -u) --build-arg gid=$(id -g) $XROAD/packages/docker-compile || errorExit "Error building build image."
+    docker run --rm -v $XROAD/..:/workspace -w /workspace/src -u builder xroad-build  bash -c "./update_ruby_dependencies.sh && ./compile_code.sh -nodaemon" || errorExit "Error running build of binaries."
```

Comment:
Should include a conditional `-it` parameter to make it possible to interrupt build with Ctrl+C, as in packaging https://github.com/nordic-institute/X-Road/blob/develop/src/build_packages.sh#L16-L18